reactor-netty-http/src/test/java/reactor/netty/http/server/forwardheaderhandler/CustomXForwardedHeadersHandler.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023 VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.server.forwardheaderhandler;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpRequest;
+import reactor.netty.http.server.ConnectionInfo;
+import reactor.netty.transport.AddressUtils;
+
+/**
+ * Custom X-Forwarded-XX headers handler which implement the same logic of the Default handler.
+ * This class is meant to verify that people can implement at least the same logic provided
+ * by the DefaultHttpForwardedHeaderHandler class.
+ */
+public final class CustomXForwardedHeadersHandler {
```

Comment:
Can we reduce the visibility? If it is not needed we don't want to expose test classes.

Reply:
The package for the new CustomXForwardedHeadersHandler (`reactor.netty.http.server.forwardheaderhandler`) is different than the one from `DefaultHttpForwardedHeaderHandler` and `ConnectionsInfoTest`.
I did that on purpose in order to make sure the custom handler is only using public ConnectionInfo API. 
So, if we reduce the visibility of CustomXForwardedHeadersHandler, then the ConnectionsInfoTest won't be able to use it ?  

Is it ok to keep the existing public visibility ? Or else, we can use some reflections tricks from ConnectionInfoTests when getting access to this class ?

Reply:
Ok let's keep this visibility, but add notice that this is not for general purpose and can be changed at any time. Is this ok?

Reply:
ok, I have updated the javadoc from CustomXForwardedHeadersHandler in order to indicate that the class is not a public API. see https://github.com/reactor/reactor-netty/pull/2789/commits/f8d975666cb2e7ddeb7b0fd747571e7455d4fa2c, can you recheck ?